CMakeLists.txt

```diff
@@ -41,31 +41,51 @@ set(UMF_PROXY_LIB_BASED_ON_POOL
 set_property(CACHE UMF_PROXY_LIB_BASED_ON_POOL
              PROPERTY STRINGS ${KNOWN_PROXY_LIB_POOLS})
 
-set(KNOWN_BUILD_TYPES Release Debug RelWithDebInfo MinSizeRel)
-string(REPLACE ";" " " KNOWN_BUILD_TYPES_STR "${KNOWN_BUILD_TYPES}")
-
-if(NOT CMAKE_BUILD_TYPE)
-    message(
-        STATUS
-            "No build type selected (CMAKE_BUILD_TYPE), defaulting to Release")
-    set(CMAKE_BUILD_TYPE "Release")
+if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+    set(LINUX TRUE)
+    set(OS_NAME "linux")
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+    set(WINDOWS TRUE)
+    set(OS_NAME "windows")
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    set(MACOSX TRUE)
+    set(OS_NAME "macosx")
 else()
-    message(STATUS "CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
-    if(NOT CMAKE_BUILD_TYPE IN_LIST KNOWN_BUILD_TYPES)
+    message(FATAL_ERROR "Unknown OS type")
+endif()
+
+# This build type check is not possible on Windows when CMAKE_BUILD_TYPE is not
+# set, because in this case the build type is determined after a CMake
+# configuration is done (at the build time)
+if(NOT WINDOWS)
```

Comment:
Perhaps for Windows we could write something - maybe simplified version of the comment above...?

Reply:
See my comment above.

Reply:
I meant, we could print some message to user :)

Reply:
With MSVC we can't print other message than always print a list of available build types. I'm not sure if such an information should be added, though.